resolved language component and added isloading spinner at fetch api

antonio-b opened the pull request:
@mikhael28 I believe I finally resolved my issues. I did console everything to make sure all the data was correct. 
Pull-Request for `paretOS`

## Description
- Changed language component into a functional while adding spinner while loading button


## Relates to
- #104  or whichever ticket number the pr is addressing

## Reviewers
- @person1 (merge duty)
- @person2
- ...


### Screenshots / other info
...

mikhael28 commented:
@antonio-b thank you so much for sending this in, I will review this tonight when I get off work. Just started a new job, yay, but that means I've been really busy the last two weeks. You buy a MacBook yet? :D

antonio-b commented:
@mikhael28  lol no, I haven't bought a new computer yet but definitely taking it under consideration. I do need to clean up the code a little on the edit profile component bc I left a unused function. I will resolve when I move on to refactoring the other functional components. It's been super busy lately for me also so thanks.

mikhael28 commented on src/profile/EditProfile.js:
```diff
@@ -409,49 +393,10 @@ export default class EditProfile extends Component {
-        />
+           <Languages  
+             id = {this.state.id}
+             user = {this.user}
```
I don't think this is correct - shouldn't it be `this.state.user`? I don't remember there being a `this.user` object in the constructor of the `EditProfile` component.

antonio-b replied:
@mikhael28  ok, those issues was resolved. you were right about the this.state.user prop being passed don't know how I didn't see that. The props.value on line 98 wasn't needed because the function updated the label to its current value. I did remove the consoles, but I will send you the GIF and another revised PR tomorrow. Thanks

mikhael28 commented on src/profile/Languages.js:
```diff
@@ -0,0 +1,137 @@
+  // The actual updated value, is represented by the event.target.value. I recommend you console.log both of the values, above the setState, so you understand.
+
+const handleChange = (e) => {
+    console.log("value: ",e.target.value);
```
Nit: remove console.log's from PR's.

FYI A 'nit' is a small comment, common dev speak.

mikhael28 commented on src/profile/Languages.js:
```diff
@@ -0,0 +1,137 @@
+            <FormControl 
+              componentClass="select"
+              onChange={handleChange}
+              value={props.value}
```
I don't see a prop being passed in called `value` - there was `id` and `user` - this doesn't seem right. With the next PR revision, would you mind uploading a GIF of the functionality working?

mikhael28 reviewed (changes requested):
Hey @antonio-b , sorry for the long delay! I've left a couple of requests, would you take a look at them? I'm not sure some of the changes are correct - would you take a look, and upload a GIF demonstrating the functionality working (the DB update being reflected in the UI on refresh)?

mikhael28 commented:
@antonio-b sometimes open-source takes time :) thanks for your patience

antonio-b commented:
No problem,  I completely understand.

antonio-b commented:
@mikhael28 ok I will look at your suggestions. I actually saw those discrepancies but everything was properly working so I assumed it was me that was wrong. I will definitely get those resolved asap. Thanks

antonio-b commented:
@mikhael28 
![p r pull](https://user-images.githubusercontent.com/53808501/144754226-1db68724-ae2c-458e-a902-37baba7fadf5.gif)

antonio-b commented:
@mikhael28  ok I made a new commit and added the gif of the u.i default language changing. I removed the consoles. and fixed the states. Everything seem to work well. I hope this is right but can and will resolve any issues you see. Sorry for the multiple commits without any resolutions.

mikhael28 reviewed (approved):
Well done! Sorry for the incredible delay, you are now merged in!

mikhael28 commented:
Well done @antonio-b - how you doing? I've been busy as hell with this new job, loving it so far. The team is amazing, the work is great - almost done building a mobile app MVP using Flutter! A nice change of pace from JS, remarkably simple language (Dart).

antonio-b commented:
@mikhael28 thanks for the merge. I completely understand about work. I just started as an associate software engineer at an airlines company. Well my title is that but my role will be an automation engineer so pretty thrilled and overwhelmed with new information.😁

mikhael28 commented:
@antonio-b that is fantastic!!! Great to hear :D what programming languages/tools are you going to be working with?

antonio-b commented:
@mikhael28 
Java language
Selenium and seetest for automation
Cucumber and jbehave tools

antonio-b commented:
@mikhael28 thanks

mikhael28 commented:
Once you get the hang of those tools, we would love for you to write some headless-browser tests with Selenium (or better yet, Cypress!). Best of luck getting ramped up :) Let me know if there is something we can help you with.

antonio-b commented:
Ok thanks. Will do.
